test-cypress/integration/explore/similarity.spec.js

```diff
@@ -0,0 +1,299 @@
+        cy.get('ul[class="nav nav-tabs"] li[class="nav-item"]').eq(1).click(); //click Search query tab
+        cy.get('.CodeMirror textarea').type('{ctrl}a{backspace}', {force: true}) //delete default search query
+        cy.get('.CodeMirror textarea').invoke('val', modifiedSearchQuery).trigger('change', {force: true});
+        cy.wait(500);
+	};
+
+    it('Test similarity page default state', function () {
+        checkSimilarityPageDefaultState();
+	});
+
+    it('Create default text similarity index', function () {
+        checkSimilarityPageDefaultState();
+        pressCreateNewIndexButton();
+        setIndexName();
+        createSimilarityIndex();
+        deleteSimilarityIndex();
+	});
+
+    it('Create default predication similarity index', function () {
+        checkSimilarityPageDefaultState();
+        pressCreateNewIndexButton();
+        selectIndexType('predication');
+        setIndexName();
+        createSimilarityIndex();
+        deleteSimilarityIndex();
+	});
+
+    it('Create text similarity index with a build parameter', function () {
+        checkSimilarityPageDefaultState();
+        pressCreateNewIndexButton();
+        setIndexName();
+        addBuildParam();
+        createSimilarityIndex();
+        deleteSimilarityIndex();
+	});
+
+    it('Create predication similarity index with a build parameter', function () {
+        checkSimilarityPageDefaultState();
+        pressCreateNewIndexButton();
+        selectIndexType('predication');
+        setIndexName();
+        addBuildParam();
+        createSimilarityIndex();
+        deleteSimilarityIndex();
+	});
+
+    it('Create text similarity index with a stop word', function () {
+        checkSimilarityPageDefaultState();
+        pressCreateNewIndexButton();
+        setIndexName();
+        addStopWord();
+        createSimilarityIndex();
+        deleteSimilarityIndex();
+	});
+
+    it('Create text similarity index with a different Lucene Analyzer', function () {
+        checkSimilarityPageDefaultState();
+        pressCreateNewIndexButton();
+        setIndexName();
+        addLuceneAnalyzer();
+        createSimilarityIndex();
+        deleteSimilarityIndex();
+	});
+
+    it('Create text similarity - literal index', function () {
+        checkSimilarityPageDefaultState();
+        pressCreateNewIndexButton();
+        setIndexName();
+        checkLiteralIndex();
+        createSimilarityIndex();
+        deleteSimilarityIndex();
+	});
+
+    it('Clone existing similarity index', function () {
+        checkSimilarityPageDefaultState();
+        pressCreateNewIndexButton();
+        setIndexName();
+        createSimilarityIndex();
+        cloneExistingIndex();
+        cy.get('button span[class="icon-trash"]').should("be.visible");
+        cy.get('button span[class="icon-trash"]').eq(0).click();
```

Comment:
Above operation can be separated in function as it's repead a couple of times.

Reply:
Changed the logic in the deleteSimilarityIndex function and added it in the clone test.